### PR TITLE
uninstall: inhibit when there are containers using the image

### PR DESCRIPTION
**Pull request by giuseppe**
## Description

do not allow to "atomic uninstall" an image if there are containers using it.

## Related Bugzillas

https://bugzilla.redhat.com/show_bug.cgi?id=1456451

### CODE REVIEW

File: Atomic/backends/_docker.py [hunk 1]
```diff
@@ -451,6 +451,12 @@ def uninstall(self, iobject, name=None, **kwargs):
 
         if args.force:
             self.delete_containers_by_image(iobject, force=True)
+        else:
+            containers_by_image = self.get_containers_by_image(iobject)
+            if len(containers_by_image) > 0:
+                containers_active = ", ".join(["- {}".format(i.name) for i in containers_by_image])
+                raise ValueError("These containers containers that are still using the image, delete them or use --force:\n"
```
Comment (rhatdan) [inline Atomic/backends/_docker.py]: +                raise ValueError("Containers %s are using this image, delete them first or use --force:\n"

File: docs/atomic-uninstall.1.md [hunk 2]
```diff
@@ -49,7 +49,7 @@ The uninstall command will not execute if --display is specified.
 If --display is not specified the uninstall command will execute.
 
 **-f** **--force**
-  Remove all containers based on this image
+  Remove all containers based on this image.  Uninstall will fail if `--force` is not used and there are still containers using the image.
```
Comment (rhatdan) [inline docs/atomic-uninstall.1.md]: Remove all containers based on this image, before removing the image.

Comment (rhatdan) [inline docs/atomic-uninstall.1.md]: Add something to the description saying that the image will not be removed if there are any containers using the image.

**Comment by giuseppe** [comment]
@rhatdan amended your suggestions and tests are passing now.

**Comment by rhatdan** [comment]
@rh-atomic-bot r+  4d2ab66

**Pull request closed without merging**